android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/RibEvents.kt

```diff
@@ -48,39 +60,48 @@ public object RibEvents {
    * for each RIB component.
    *
    * @param ribAction The related RIB action type. e.g. didBecomeActive, willLoad, etc
-   * @param ribCallerClassType Related RIB component class type
+   * @param ribComponent Related RIB component
    * @param ribComponentType The RIB component type (e.g. Interactor, Router, Presenter, Worker)
    * @param ribEventType RIB event type (e.g. ATTACH/DETACH)
    */
-  internal inline fun <T : Any> triggerRibActionAndEmitEvents(
-    ribCallerClassType: T,
+  internal inline fun triggerRibActionAndEmitEvents(
+    ribComponent: RibComponent,
     ribComponentType: RibComponentType,
```

Comment:
I would like to specify a property on the new RibComponent to be overriden, unfortunately given that Worker and RibCoroutineWorker are interfaces we cannot restrict visibility of those